server-go/filtering/parser/parser.go

```diff
@@ -0,0 +1,417 @@
+	if ctx.GetOpen() == nil {
+		return p.newIdent(ctx, id)
+	}
+	var args []*expr.Expr = nil
+	if ctx.ArgList() != nil {
+		args = p.Visit(ctx.ArgList()).([]*expr.Expr)
+	}
+	return p.newCall(ctx.GetOpen(), id, args)
+}
+
+// Return a list of Expr values to be used as arguments.
+func (p *parser) VisitArgList(ctx *gen.ArgListContext) interface{} {
+	exprArgs := make([]*expr.Expr, len(ctx.GetArgs()))
+	for i, arg := range ctx.GetArgs() {
+		exprArgs[i] = p.Visit(arg).(*expr.Expr)
+	}
+	return exprArgs
+}
+
+// Return a string field name for use with a selected field or qualified
+// function.
+func (p *parser) VisitField(ctx *gen.FieldContext) interface{} {
+	if ctx.Keyword() != nil {
+		return p.Visit(ctx.Keyword())
+	}
+	if ctx.GetQuotedText() != nil {
+		return p.unquote(ctx, ctx.GetQuotedText().GetText())
+	}
+	return p.Visit(ctx.GetId())
+}
+
+// Return a string constant value.
+// Different filter consumers may choose to support conventions for converting
+// a string to a Timestamp or Duration. This parser does not attempt any more
+// intelligent interpretation of the literal.
+func (p *parser) VisitStringVal(ctx *gen.StringValContext) interface{} {
+	text := ctx.GetText()
+	return p.newConst(ctx, p.unquote(ctx, text))
+}
+
+func (p *parser) VisitNumber(ctx *gen.NumberContext) interface{} {
+	if ctx.FloatVal() != nil {
+		return p.Visit(ctx.FloatVal())
+	}
+	return p.Visit(ctx.IntVal())
+}
+
+// Return an int64 value from the parsed string.
+func (p *parser) VisitIntVal(ctx *gen.IntValContext) interface{} {
+	text := ctx.GetText()
+	val, err := strconv.ParseInt(text, 0, 64)
+	if err == nil {
+		return p.newConst(ctx, val)
+	}
+	p.errors.ReportError(
+		p.source,
+		common.Location(ctx.GetStart()),
+		fmt.Sprintf("Unrecognized integer value: %s", text))
+	return p.newConst(ctx, "<<error>>")
+}
+
+// Return a float64 value from the parsed string.
+func (p *parser) VisitFloatVal(ctx *gen.FloatValContext) interface{} {
+	text := ctx.GetText()
+	val, err := strconv.ParseFloat(text, 64)
+	if err == nil {
+		return p.newConst(ctx, val)
+	}
+	p.errors.ReportError(
+		p.source,
+		common.Location(ctx.GetStart()),
+		fmt.Sprintf("Unrecognized floating point value: %s", text))
+	return p.newConst(ctx, "<<error>>")
+}
+
+// Listener implementations
+func (p *parser) SyntaxError(recognizer antlr.Recognizer,
+	offendingSymbol interface{}, line, column int, msg string,
+	e antlr.RecognitionException) {
+	var errorMsg = "Syntax error"
+	switch e.(type) {
+	case antlr.InputMisMatchException:
+		errorMsg = "Input mismatch"
+	case antlr.NoViableAltException:
+		errorMsg = "Unexpected token"
+	}
+	p.errors.ReportError(p.source, common.NewLocation(line, column), errorMsg)
+}
+
+func (p *parser) ReportAmbiguity(recognizer antlr.Parser, dfa *antlr.DFA,
+	startIndex, stopIndex int, exact bool, ambigAlts *antlr.BitSet,
+	configs antlr.ATNConfigSet) {
+	// Intentional
+}
+
+func (p *parser) ReportAttemptingFullContext(recognizer antlr.Parser,
```

Comment:
Same here and below with function level comments.

Reply:
Of course, happy to add more comments.